src/vs/workbench/parts/files/electron-browser/fileActions.ts

```diff
@@ -1335,35 +1336,77 @@ export function validateFileName(parent: IFileStat, name: string, allowOverwriti
 		return nls.localize('emptyFileNameError', "A file or folder name must be provided.");
 	}
 
+	const names: string[] = trimTrailingSlashes(name)	// prevents empty last array element after split
```

Comment:
Remove `trimTrailingSlashes` ,just split go through the array and filetr out null elements. Like this
`name.split(/[\\/]/).filter(part => !!part);`
imho that is simpler